### PR TITLE
[IMP] website_sale: Update attribute display type in front

### PR DESCRIPTION
**Pull request by Williambraecky**
Allow editing the attribute's display type from the front end using the
web editor.

TaskId-2675390

### CODE REVIEW

File: addons/sale/views/variant_templates.xml [hunk 1]
```diff
@@ -17,7 +18,7 @@
                     <t t-if="ptal.attribute_id.display_type == 'select'">
                         <select
                             t-att-data-attribute_id="ptal.attribute_id.id"
-                            t-attf-class="custom-select css_attribute_select js_variant_change #{ptal.attribute_id.create_variant} #{'d-none' if single_and_custom else ''}"
+                            t-attf-class="custom-select css_attribute_select product_attribute js_variant_change #{ptal.attribute_id.create_variant} #{'d-none' if single_and_custom else ''}"
```
Comment (rdeodoo) [inline addons/sale/views/variant_templates.xml:21]:
```suggestion
                            t-attf-class="custom-select css_attribute_select o_wsale_product_attribute js_variant_change #{ptal.attribute_id.create_variant} #{'d-none' if single_and_custom else ''}"
```
Please try to follow the convention, it makes it easier to figure which classes are related to odoo.
I am not too sure of what it exactly is, `o_wsale_` or `o_` or `oe_`, but at least something to better identify it

(You might wanna change it now that you handle the module, but probably keep something in that regarde)

**Review by rdeodoo — COMMENTED** [review]
Could be of interest :)
Clicked on mail notif out of curiosity

**Review by tlibioulle — APPROVED** [review]
LGTM :)

I think it was ok for @rdeodoo too ?

Thanks for your work !

**Comment by rdeodoo** [comment]
@tlibioulle It was just a random remark passing by and it got applied so it's fine indeed :)

File: addons/website_sale/static/src/js/website_sale.editor.js [hunk 2]
```diff
@@ -682,4 +682,39 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
...
 });
+
+options.registry.WebsiteSaleProductAttribute = options.Class.extend({
+    /**
+     * @override
+     */
+     willStart: async function () {
+        const _super = this._super.bind(this);
+        this.attributeID = this.$target.closest('[data-attribute_id]').data('attribute_id');
+        return _super(...arguments);
```
Comment (tlibioulle) [inline addons/website_sale/static/src/js/website_sale.editor.js:693]: I know the code comes from the WebsiteSaleProductsItem but I wonder if you cannot write it like this :

```suggestion
        this.attributeID = this.$target.closest('[data-attribute_id]').data('attribute_id');
        return this._super(...arguments);
```

File: addons/website_sale/static/src/js/website_sale.editor.js [hunk 3]
```diff
@@ -682,4 +682,39 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
...
+     * @see this.selectClass for params
+     */
+    setDisplayType: function (previewMode, widgetValue, params) {
+        this._rpc({
+            route: '/shop/config/attribute',
+            params: {
+                attribute_id: this.attributeID,
+                display_type: widgetValue,
+            },
+        }).then(reload);
```
Comment (tlibioulle) [inline addons/website_sale/static/src/js/website_sale.editor.js:706]: I wonder if it wouldn't be better to display a message below the <we-select> to tell the user the modification will be visible only after the reload ?

I think the drawback of losing other changes when you're encouraged to reload the page is maybe too strict and not enough intuitive.

**Review by tlibioulle — COMMENTED** [review]
~~Here is another point-of-view regarding the select-and-reload behavior which discards current changes of the editor.~~

~~Wouldn't it be better to have maybe a softer way to tell the user the change was saved but will only be applied after a reload ?
Regarding UX, I think the user will have the same numbers of click (clicking on Save) and he won't have the reload alert message.~~

[EDIT] As it was validated functionally and it works like numerous other options, I think there is no point here.

We can go ahead.

Otherwise, same comment applies as my first eye on it, clean&nice work.

Thanks a lot !

File: addons/website_sale/static/src/js/website_sale.editor.js (hunk 3 above)
Comment (tlibioulle) [reply addons/website_sale/static/src/js/website_sale.editor.js:706]: Side note: As it was validated functionally and it works like that for numerous other options, I think there is no point here.

We can go ahead.

Comment (Williambraecky) [reply addons/website_sale/static/src/js/website_sale.editor.js:706]: It follows the behaviour of some other changes we can do with the web editor, I think's it's ok as is.

**Comment by tlibioulle** [comment]
@tivisse :rocket:

**Comment by tivisse** [comment]
@robodoo r+

**Comment by tivisse** [comment]
@robodoo r+

**Pull request closed without merging**